libs/core/src/components/datepicker/datepicker.test.ts

```diff
@@ -513,5 +514,34 @@ describe('<gds-datepicker>', () => {
         onlyDate(new Date('2024-01-01')),
       )
     })
+
+    it('should not overflow the year when trying to increase max year', async () => {
```

Comment:
It would be good to test the previous month action too. That could be done in a separate test

Reply:
Fixed